src/services/conversation.service.ts

```diff
@@ -7,22 +7,14 @@ import { Conversation } from '@entities/conversation.entity';
 export class ConversationsService {
   private readonly conversationRepository = getRepository<Conversation>(Conversation);
 
-  listConversations() {
-    return this.conversationRepository.find();
+  listConversationsByUser(userId: number) {
+    return this.conversationRepository.find({ where: [{ user1Id: userId }, { user2Id: userId }] });
   }
 
-  // showTopic(id: number) {
-  //   return this.conversationRepository.findOne(id);
-  // }
-
   async createConversation(conversation: Conversation) {
     const exists = (await this.conversationRepository.count({ where:
       [{ user1Id: conversation.user1Id, user2Id: conversation.user2Id },
         { user1Id: conversation.user2Id, user2Id: conversation.user1Id }] })) > 0;
     return !exists? this.conversationRepository.save(conversation) : null;
   }
```

Comment:
I suggest breaking this function in two: one that checks if the conversation exists and another one to create the conversation, because here we are breaking the single responsability pattern ( It is checking if the conversation exists and also is creating the conversation). You can have, for example:

`async checkIfConversationExists: Promise<boolean>` and `async createConversation: Promise<Conversation>`

Reply:
I agree, thank you.